sympy/functions/special/beta_functions.py

```diff
@@ -118,3 +129,227 @@ def _eval_conjugate(self):
+
+    def _eval_rewrite_as_hyper(self, a, b, x1, x2, **kwargs):
+        from sympy.functions.special.hyper import hyper
+        return (x2**a * hyper((a, 1 - b), (a + 1,), x2) - x1**a * hyper((a, 1 - b), (a + 1,), x1)) / a
+
+###############################################################################
+#################### REGULARIZED INCOMPLETE BETA FUNCTION #####################
+###############################################################################
+
+class betainc_regularized(Function):
+    r"""
+    The Generalized Regularized Incomplete Beta function is given by
+
+    .. math::
+        \mathrm{I}_{(x_1, x_2)}(a, b) = \frac{\mathrm{B}_{(x_1, x_2)}(a, b)}{\mathrm{B}(a, b)}
+
+    The Regularized Incomplete Beta function is a special case
+    of the Generalized Regularized Incomplete Beta function :
+
+    .. math:: \mathrm{I}_z (a, b) = \mathrm{I}_{(0, z)}(a, b)
+
+    The Regularized Incomplete Beta function is the cumulative distribution
+    function of the beta distribution.
+
+    Examples
+    ========
+
+    >>> from sympy import betainc_regularized, symbols, conjugate
+    >>> a, b, x, x1, x2 = symbols('a b x x1 x2')
+
+    The Generalized Regularized Incomplete Beta
+    function is given by:
+
+    >>> betainc_regularized(a, b, x1, x2)
+    betainc_regularized(a, b, x1, x2)
+
+    The Regularized Incomplete Beta function
+    can be obtained as follows:
+
+    >>> betainc_regularized(a, b, 0, x)
+    betainc_regularized(a, b, 0, x)
+
+    The Regularized Incomplete Beta function
+    obeys the mirror symmetry:
+
+    >>> conjugate(betainc_regularized(a, b, x1, x2))
+    betainc_regularized(conjugate(a), conjugate(b), conjugate(x1), conjugate(x2))
+
+    We can numerically evaluate the Regularized Incomplete Beta
+    function to arbitrary precision on the whole complex plane:
+
+    >>> from sympy import betainc_regularized, pi, E
+    >>> betainc_regularized(1, 2, 0, 0.25).evalf(10)
+    0.4375000000
+    >>> betainc_regularized(pi, E, 0, 1).evalf(5)
+    1.00000
+
+    The Generalized Regularized Incomplete Beta function can be
+    expressed in terms of the Generalized Hypergeometric function.
+
+    >>> from sympy import hyper
+    >>> betainc_regularized(a, b, x1, x2).rewrite(hyper)
+    (-x1**a*hyper((a, 1 - b), (a + 1,), x1) + x2**a*hyper((a, 1 - b), (a + 1,), x2))/(a*beta(a, b))
+
+    See Also
+    ========
+
+    beta: Beta function
+    hyper: Generalized Hypergeometric function
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Beta_function#Incomplete_beta_function
+    .. [2] https://dlmf.nist.gov/8.17
+    .. [3] https://functions.wolfram.com/GammaBetaErf/Beta4/
+    .. [4] https://functions.wolfram.com/GammaBetaErf/BetaRegularized4/02/
+
+    """
+    nargs = 4
+    unbranched = True
+
+    def __new__(cls, a, b, x1, x2):
+        return Function.__new__(cls, a, b, x1, x2)
+
+    def _eval_mpmath(self, func, args):
+        return func(*args, 1)
```

Comment:
I wonder if this should actually take a prec argument and try to recursively call evalf

Reply:
Maybe something like this would be better:
```
_eval_mpmath = getattr(self, '_eval_mpmath', None)
if _eval_mpmath is None:
     func = _get_mpmath_func(self.func.__name__)
else:
     func, args = _eval_mpmath()
```

Reply:
The reason I went with this type of syntax is so that we have complete freedom over how the args are rearranged. For example, if I wanted to change the API of `betainc_regularized` from `betainc_regularized(a, b, x, y)` to `betainc_regularized(x, y, a, b)`, all I'd have to do is change the definition in `__new__` and in `_eval_mpmath`
```
def _eval_mpmath(self, func, args):
    return func(args[1], args[3], args[0], args[1], 1)
```
This also solves the problem of the hidden args.

Reply:
I was thinking that it might be good to for `_eval_mpmath` to be able to refer to a completely different mpmath function. I suppose it can (if it ignores the `func` argument)...

Reply:
But would that not mean that we have to import mpmath functions in the sympy codebase apart from them being imported in core? Could you give an example? I'm not sure if I correctly understood what you said

Reply:
Yes, it would mean that. That's fine though because mpmath is a hard dependency. It is good to keep the imports in one place if possible but we also want to make `evalf` extensible from outside.

Reply:
Ok, thanks for clarifying. I'll change this and import the function directly